snapcraft/internal/elf.py

```diff
@@ -198,7 +205,14 @@ def _extract(self, path) -> Tuple[str, str, Dict[str, NeededLibrary]]:
                     lib = libs[library_name]
                     for version in versions:
                         lib.add_version(_ensure_str(version.name))
-        return interp, soname, libs
+
+            for segment in elf.iter_segments():
+                if segment['p_type'] == 'PT_GNU_STACK':
+                    mode = segment['p_flags']
+                    if mode & 1:
+                        execstack_set = True
```

Comment:
I think this would be clearer with:
```
# p_flags holds the bit mask for this segment. See `man 5 elf`
if mode & elftools.elf.constants.P_FLAGS.PF_X:
    execstack_set = True
```

Reply:
thanks @jdstrand, I've just pushed fixes with your recommendations with an integration test for completenes